source/dub/generators/generator.d

```diff
@@ -435,7 +441,61 @@ class ProjectGenerator
 		configureDependents(*roottarget, targets);
 		visited.clear();
 
-		// 4. Filter applicable version and debug version identifiers
+		// 4. As an extension to configureDependents we need to copy any finalBinarySourceFile
+		// in our dependencies (ignoring targetType)
+		void configureDependentsFinalImages(TargetInfo* ti, TargetInfo[string] targets, TargetInfo* ifRootIsFinalImage = null, size_t level = 0)
```

Comment:
I realized after trying to get some sleep that this function isn't quite as recursive as I thought it would be.

``ifRootIsFinalImage`` should probably be an array.

Reply:
I've been thinking about this further, with this fixed a CI test actually wouldn't pass, due to issues surrounding dynamicLibraries. So that'll need to be fixed before this is pulled.